src/components/main/favorite-button/favorite-button.tsx

```diff
@@ -1,22 +1,53 @@
 import clsx from 'clsx';
-import { BemClass } from '../../../const';
+import { AppRoute, AuthStatus, BemClass } from '../../../const';
+import { AllOffersType } from '../../../types/offer-type';
+import { dispatch } from '../../../store/store';
+import {
+  fetchOffersAction,
+  fetchSetFavoriteAction,
+} from '../../../store/api-actions/offers-actions';
+import { useAppSelector } from '../../../hooks/store';
+import { userSelectors } from '../../../store/slices/user-slice';
+import { useNavigate } from 'react-router-dom';
 
 type FavoriteButtonProps = {
   bemBlock: BemClass;
   isFavorite: boolean;
+  currentOffer: AllOffersType;
 };
 
-function FavoriteButton({ bemBlock, isFavorite }: FavoriteButtonProps) {
+function FavoriteButton({
+  bemBlock,
+  isFavorite,
+  currentOffer,
+}: FavoriteButtonProps) {
+  const navigate = useNavigate();
+  const authStatus = useAppSelector(userSelectors.status);
+  const isAuth = authStatus === AuthStatus.Auth;
   const favoriteLabel = `${isFavorite ? 'In' : 'To'} bookmarks`;
   const favoriteClasses = clsx('button', `${bemBlock}__bookmark-button`, {
     [`${bemBlock}__bookmark-button--active`]: isFavorite,
   });
 
+  const handleFavoriteButtonClick = (offer: AllOffersType) => {
+    if (isAuth) {
+      const status: 1 | 0 = offer.isFavorite ? 0 : 1;
+      dispatch(fetchSetFavoriteAction({ offer, status }));
+      dispatch(fetchOffersAction());
```

Comment:
потому что ты эти две санки (промиса) диспатчишь одновременно, а не ждёшь пока один обновится. И в итоге у тебя race condition
```suggestion
  const handleFavoriteButtonClick = async (offer: AllOffersType) => {
    if (isAuth) {
      const status: 1 | 0 = offer.isFavorite ? 0 : 1;
      await dispatch(fetchSetFavoriteAction({ offer, status })).unwrap();
      await dispatch(fetchOffersAction()).unwrap();
```

Хотя я бы все равно рекомендовал бы какое-то другое решение